src/IO/ProcessRunner.php

```diff
@@ -13,4 +13,6 @@
      * @return Process
      */
     public function run(Process $process, $mustSucceed = true);
+
+    public function setUserInteraction(UserInteraction $userInteraction);
```

Comment:
I don't want to overcomplicate this and avoiding introducing factories would be good. Pushing the issue to the UserInteraction class would help as that could then check that the Output dependency has been set before it is used in a single place. It means that you can have an exception instead of a method called on a non object fatal which at least helps with debugging.